### PR TITLE
size presubmit check

### PR DESCRIPTION
**Pull request by Feiyang1**

### CODE REVIEW

File: scripts/report_binary_size.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
+const { resolve } = require('path');
+const fs = require('fs');
+const { execSync } = require('child_process');
+
+const repoRoot = resolve(__dirname, '..');
+
+const commitHash = process.env.GITHUB_SHA || execSync('git rev-parse HEAD').toString();
```
Comment (yifanyang) [inline scripts/report_binary_size.js]: We seem to use `simple-git` for git operations in [other places in this repo](https://github.com/firebase/firebase-js-sdk/blob/master/scripts/release/utils/git.js#L18).

I'm okay with both. Will leave it up to you whether we want to pursue any consistency.

File: scripts/report_binary_size.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    
+    const files = [
+        ...special_files.map(file => `${firebaseRoot}/${file}`), 
+        ...pkgJson.components.map(component => `${firebaseRoot}/firebase-${component}.js`)
+    ];
+    
+    for (const file of files) {
+        const { size } = fs.statSync(file);
+        const fileName = file.split('/').slice(-1)[0]
+        reports.push(makeReportObject('firebase', fileName, size))
```
Comment (yifanyang) [inline scripts/report_binary_size.js]:
```suggestion
        const fileName = file.split('/').slice(-1)[0];
        reports.push(makeReportObject('firebase', fileName, size));
```

File: scripts/report_binary_size.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+    return reports;
+}
+
+// @firebase/*
```
Comment (yifanyang) [inline scripts/report_binary_size.js]:
```suggestion
// NPM packages
```

File: scripts/report_binary_size.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        ...generateReportForCDNScripts(),
+        ...generateReportForNPMPacakges()
+    ];
+
+    for (const r of reports) {
+        console.log(r.sdk, r.type, r.value);
+    }
+
+    return {
+        log: "https://www.goog.com",
```
Comment (yifanyang) [inline scripts/report_binary_size.js]: I guess this is only applicable to CI runs. So maybe we want to configure it to be some value when on CI and empty when running locally.

Or we can leave this field empty. The only use of this field is to provide a link on the PR comment for developers to examine the original log (when they are surprised by the number.)

File: scripts/report_binary_size.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        metric: "BinarySize",
+        results: reports
+    };
+}
+
+generateSizeReport();
```
Comment (yifanyang) [inline scripts/report_binary_size.js]: Can we write the result JSON to a file? The way to connect it to the metrics service to make a http post request with that file as the request body.

Comment (yifanyang) [reply scripts/report_binary_size.js]: Oh, never mind. We are probably going to sent request programmatically anyway. So it is good now.

File: scripts/report_binary_size.js [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+        const { size } = fs.statSync(file);
+        const fileName = file.split('/').slice(-1)[0]
+        reports.push(makeReportObject('firebase', fileName, size))
+    }
+
+    return reports;
+}
+
+// @firebase/*
+function generateReportForNPMPacakges() {
```
Comment (hsubox76) [inline scripts/report_binary_size.js]: Typo ("Pacakges")

File: scripts/report_binary_size.js (hunk 2 above)
Comment (Feiyang1) [reply scripts/report_binary_size.js]: Done.

File: scripts/report_binary_size.js (hunk 3 above)
Comment (Feiyang1) [reply scripts/report_binary_size.js]: Done.

File: scripts/report_binary_size.js (hunk 6 above)
Comment (Feiyang1) [reply scripts/report_binary_size.js]: Corrected.

File: scripts/report_binary_size.js (hunk 4 above)
Comment (Feiyang1) [reply scripts/report_binary_size.js]: updated to report github action url.

**Comment by Feiyang1** [comment]
replaced by https://github.com/firebase/firebase-js-sdk/pull/2744

**Pull request closed without merging**